src/tennisSimulator/MatchSimulator.java

```diff
@@ -29,10 +36,118 @@ public void changeServer() {
     }
 
     public void showResult() {
-        System.out.println("\t\t  GAME \t SET \tMATCH");
-        System.out.println(firstPlayer.getAbbreviatedName() + "  " + firstPlayer.getHandsWon() + "\t   "
-                + firstPlayer.getGamesWon() + "\t  " + firstPlayer.getSetsWon());
-        System.out.println(secondPlayer.getAbbreviatedName() + "  " + secondPlayer.getHandsWon() + "\t   "
-                + secondPlayer.getGamesWon() + "\t  " + secondPlayer.getSetsWon());
+        if (noWinner()) {
+            if (firstPlayer.isServing()) {
+                System.out.println(firstPlayer.getName() + " is serving");
+            } else {
+                System.out.println(secondPlayer.getName() + " is serving");
+            }
+            System.out.println("\t\t  GAME \t SET");
+            System.out.println(firstPlayer.getAbbreviatedName() + "  " + convertHandsScore(firstPlayer.getHandsWon())
+                    + "\t   " + firstPlayer.getGamesWon() + "\t  " + firstPlayer.getSetsWon());
+            System.out.println(secondPlayer.getAbbreviatedName() + "  " + convertHandsScore(secondPlayer.getHandsWon())
+                    + "\t   " + secondPlayer.getGamesWon() + "\t  " + secondPlayer.getSetsWon());
+        } else {
+            System.out.println("Match is finished. Match result: ");
+            System.out.print("\t");
+            for (String result : setResults.values()) {
+                System.out.print(result + " ");
+            }
+            System.out.println("\n"+ "[" + firstPlayer.getRating() + "]" + firstPlayer.getName() + " "
+                    + firstPlayer.getSetsWon() + " - " + secondPlayer.getSetsWon() + " " + secondPlayer.getName()
+                    + "[" + secondPlayer.getRating() + "]\n");
+            resetValues();
+        }
+    }
+
+    public void simulateHand() {
+        double ratingDifference = firstPlayer.getRating() - secondPlayer.getRating();
+        double servingFactor = 0.2;
+        if (firstPlayer.isServing()) {
+            servingFactor = -0.2;
+        }
+        double probability = 1 / (1 + Math.pow(10, ((ratingDifference / 150) * 0.8 + servingFactor)));
+        applySimulationResult(probability);
+    }
+
+    public void updateResult() {
+        if (firstPlayer.getHandsWon() == 4 && secondPlayer.getHandsWon() < 3 || firstPlayer.getHandsWon() == 5) {
```

Comment:
All those checks should be in self explaining method, because it is hard to understand all those number comparisons.